tests/NCronJob.Tests/NCronJobIntegrationTests.cs

```diff
@@ -200,7 +200,8 @@ public async Task ThrowIfJobWithDependenciesIsNotRegistered()
 
         await Assert.ThrowsAsync<InvalidOperationException>(async () =>
         {
-            using var executor = provider.CreateScope().ServiceProvider.GetRequiredService<JobExecutor>();
+            using var serviceScope = provider.CreateScope();
+            using var executor = serviceScope.ServiceProvider.GetRequiredService<JobExecutor>();
```

Comment:
Good catch. Was this contributing to the flakyness?

Reply:
Unfortunately not at all - it is still flaky. The problem is that as long as you have a somewhat beefy CPU, you will not notice.
Locally I used the "Run until failure" feature for multiple minutes and didn't encounter a single faulty run - two runs onto GitHub actions and I got the first red test.

That indicates at least some race condition.